modules/serverless-deploy-func-kn.adoc

```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// serverless/serverless-functions-quickstart.adoc
+// add user guide?
+
+[id="serverless-deploy-func-kn_{context}"]
+= Deploying functions
+
+You can deploy a function to your cluster as a Knative service, by using the `kn func deploy` command.
+
+If the targeted function is already deployed, it is updated with a new container image that is pushed to a container image registry, and the Knative service is updated.
+
+[NOTE]
+====
+If you are using `quay.io` as the image registry, you must ensure that the repository is not private. The first time an image is deployed these settings may need to be updated.
+====
+
+.Prerequisites
+
+* You must have already initialized the function that you want to deploy.
+
+.Procedure
+
+* Deploy a function:
++
+[source,terminal]
+----
+$ kn func deploy [-n <namespace> -p <path> -i <image> -r <registry>]
+----
+** If no `namespace` is specified, the function will be deployed in the current namespace.
+** The function is deployed from the current directory, unless a `path` is specified.
+** The Knative service name is derived from the project name, and cannot be changed using this command.
```

Comment:
I think it would be nice to have example command and example output as well.

Reply:
@lance can you provide this please?

Reply:
Regarding output, the CLI overwrites each line of output with the next line unless in verbose mode (using the `-v` flag). So, typical output would simply be:

```console
$ kn func deploy
Function deployed at: http://func.my.cluster.com
```